councilmatic_core/management/commands/convert_attachment_text.py

```diff
@@ -0,0 +1,118 @@
+                          server_side_cursors=True)
+
+class Command(BaseCommand):
+    help = 'Converts bill attachments into plain text'
+    
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--update_all',
+            default=False,
+            action='store_true',
+            help='Add or update plain text for all bill attachments.')
+
+    def handle(self, *args, **options):
+        self.update_all = options['update_all']
+        self.connection = engine.connect()
+        self.add_plain_text()
+
+    def get_document_url(self):
+        self.connection.execute("SET local timezone to '{}'".format(settings.TIME_ZONE))
+        with engine.begin() as connection:
+            # Only apply this query to most recently updated (or created) bill documents.
+            max_updated = BillDocument.objects.all().aggregate(Max('updated_at'))['updated_at__max']
+
+            if max_updated is None or self.update_all:
+                query = '''
+                    SELECT id, url
+                    FROM councilmatic_core_billdocument 
+                    WHERE document_type='A' 
+                    AND full_text is null
+                    AND lower(url) similar to '%(.doc|.docx|.pdf)'
+                    ORDER BY updated_at DESC
+                ''' 
+            else:
+                query = '''
+                    SELECT id, url
+                    FROM councilmatic_core_billdocument 
+                    WHERE updated_at >= :max_updated
+                    AND document_type='A' 
+                    AND full_text is null
+                    AND lower(url) similar to '%(.doc|.docx|.pdf)'
+                    ORDER BY updated_at DESC
+                '''
+
+            result = connection.execution_options(stream_results=True).execute(sa.text(query), max_updated=max_updated)
+
+            yield from result
+
+    def convert_document(self):
+        documents = self.get_document_url()
+
+        logger.info('Converting document to plain text...')
+
+        for document_data in documents:
+            document_data = dict(document_data)
+            url = document_data['url']
+            document_id = document_data['id']
+            response = requests.get(url)
+            extension = os.path.splitext(url)[1]
+
+            with tempfile.NamedTemporaryFile(suffix=extension) as tfp:
+                tfp.write(response.content)
+                plain_text = textract.process(tfp.name)
+
+            yield {'plain_text': plain_text.decode('utf-8'), 'id': document_id}
+            
+
+    def add_plain_text(self):
+        plain_text_results = self.convert_document()
+
+        self.connection.execute("SET local timezone to '{}'".format(settings.TIME_ZONE))
+        query = '''
+            UPDATE councilmatic_core_billdocument as bill_docs
+            SET full_text = :plain_text
+            WHERE bill_docs.id = :id  
+        '''
+
+        chunk = []
+
+        for doc_dict in plain_text_results:
```

Comment:
It'd be a bit more readable here to do `for doc_dict in self.convert_document()` rather than assigning the generator function call to a variable, I think.

Reply:
Could you be a bit more descriptive with the `convert_document` name, perhaps `convert_document_to_plaintext`?